### PR TITLE
Added OAuth for Comet

### PR DESCRIPTION
**Pull request by ThomasPoot**
…are provided, using a new error type for the event of a failure.

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java [hunk 1]
```diff
@@ -1,33 +1,80 @@
...
+      this.auth = future.get();
+    } catch (MalformedURLException | InterruptedException | ExecutionException e) {
+      throw new OAuthFailedException(e);
+    } finally {
+      service.shutdown();
+    }
   }
 
-  public void sendCreateJobRequest(CaseRequest caseRequest, String caseId) {
+  public void sendCreateJobRequest(CaseRequest caseRequest, String caseId) throws OAuthFailedException {
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: catch the OauthFailedException and wrapped it with a GatewayException. see GatewayActionsReceiver.convertMessageToDTO(...) as an example. 
I find having a lost list of exception propagating up is ugly and confusing, as its not to the extreme of some practices of only throwing Runtime exceptions.

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
+    if (!isAuthed() && !clientID.isEmpty() && !clientSecret.isEmpty())
+      auth();
     JobServiceUtils.printJSON(caseRequest);
-    HttpEntity<CaseRequest> body = new HttpEntity<>(caseRequest);
+    HttpHeaders httpHeaders = new HttpHeaders();
+    if (isAuthed())
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: Im pretty sure the toke has a life span of an hour, you need to check the tokens expiry date(time) against current date and acquire a refresh token(if you can get that working, some post say it doesnt work) or get a new token.

Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: This has been added!

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java (hunk 1 above)
Comment (ThomasPoot) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/rest/client/CometRestClient.java]: This has been added!

**Pull request merged**